client/sftp.js

```diff
@@ -0,0 +1,83 @@
+import sftp from 'ssh2-sftp-client';
+
+export class SFTPClient {
+  constructor(timeout, logger, encoding) {
+    this.encoding = encoding;
+    this.timeout = timeout;
+    this.log = logger;
+
+    this.readOptions = {
+      readStreamOptions: {
+        flags: 'r',
+        encoding: this.encoding,
+      },
+      pipeOptions: {
+        end: false
+      }
+    };
+
+    this.client = new sftp();
+
+    this.connectionClosed = true;
+
+    this.client.on('close', this.closedConnection);
+    this.client.on('end', this.closedConnection);
+  }
+
+  async size(path) {
+    if (this.connectionClosed) throw Exception('Connect first in order to use this function');
```

Comment:
connectionClosed cannot be removed since it is being used when the client tries to connect. Since I couldn't get a state of the connection I handle it with this variable.

Reply:
I think we can still remove this error though, which you've done - I'll resolve this conversation.